torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -686,6 +683,51 @@ TensorView* castIntermediateValueInCompleteFusion(
   return fp16_tv;
 }
 
+TensorView* castIntermediateValueInCompleteFusionBF16(
```

Comment:
Confirmed with @kevinstephano that:
1. cast policy for half and bf are the same
2. we will not have a case where the two are mixed. 

So instead of having two copies of parallel logic for fp16 and bf, I'd strongly prefer adding an argumet of type `DataType`. Renamings of member function/variables are nice after this extension.